payments_service/auth/tests/test_views.py

```diff
@@ -30,6 +31,12 @@ def set_solitude_buyer(self):
         self.solitude.generic.buyer.get_object.return_value = buyer
         return buyer
 
+    def set_solitude_buyer_patcher(self):
+        # Set up support for solitude.generic.buyer(user_id).patch(...)
+        buyer = mock.Mock()
+        self.solitude.generic.buyer.return_value = buyer
+        return buyer.patch
```

Comment:
yeah, I know what you mean. It always feels awkward. I only see a couple issues with mocks: 1) when the upstream API changes you have to change the mocks and 2) they don't exercise real code, just a contractual obligation. For 1), the upstream problem, putting the mock definition in *one* place (like these setup methods) rather than multiple places addresses it. For 2), Python doesn't have good support for contracts so you really need end to end tests to cover it well.